### PR TITLE
Use ko to build container image

### PR DESCRIPTION
**Pull request by scothis**
`ko` is a tool that can build and deploy k8s resources that reference
go paths.

https://github.com/google/ko

ko works by finding `ko://` references in the k8s yaml that point at a
go compile target, the built binary is placed into the base image and
pushed to an image registry specified by the KO_DOCKER_REPO env var. A
digested reference to the pushed image is substituted into the yaml
replacing the `ko://...` reference.

In this case, we can replace direct dependencies on the kubectl and
docker CLIs and the Docker daemon. ko also makes it much easier to
produce multi-platform/architecture images for which there is a
compatible base image (gcr.io/distroless/static:nonroot by default) and
can be cross compiled by go.

Signed-off-by: Scott Andrews <andrewssc@vmware.com>

### CODE REVIEW

File: config/crd/bases/service.binding_servicebindings.yaml [hunk 1]
```diff
@@ -4,7 +4,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.4.1
+    controller-gen.kubebuilder.io/version: v0.6.2
```
Comment (baijum) [inline config/crd/bases/service.binding_servicebindings.yaml:7]: This is an unrelated change, right?
I think we need a check in the CI to ensure `make generate` and `make manifests` commands are run.
For now, let this change go with this PR.

File: Makefile [hunk 2]
```diff
@@ -67,23 +63,16 @@ build: generate fmt vet ## Build manager binary.
 run: manifests generate fmt vet ## Run a controller from your host.
 	go run ./main.go
 
-docker-build: test ## Build docker image with the manager.
-	docker build -t ${IMG} .
```
Comment (baijum) [inline Makefile:71]: Do we need an equivalent `ko` command to build an image here?

File: Makefile [hunk 3]
```diff
@@ -67,23 +63,16 @@ build: generate fmt vet ## Build manager binary.
...
-
-docker-push: ## Push docker image with the manager.
-	docker push ${IMG}
```
Comment (baijum) [inline Makefile:74]: Do we need an equivalent `ko` command to push the image here?

File: config/crd/bases/service.binding_servicebindings.yaml (hunk 1 above)
Comment (scothis) [reply config/crd/bases/service.binding_servicebindings.yaml:7]: exactly, 👍

File: Makefile (hunk 2 above)
Comment (scothis) [reply Makefile:71]: `ko apply` will:
- build images
- publish images to the registry
- substitute the image into the yaml
- apply the yaml to the cluster

File: Makefile (hunk 3 above)
Comment (scothis) [reply Makefile:74]: https://github.com/k8s-service-bindings/implementation/pull/19#discussion_r692268506

File: Makefile (hunk 2 above)
Comment (baijum) [reply Makefile:71]: Ok :+1: 

I am new to this tool. Thanks for the explanation.

**Review by baijum — APPROVED** [review]
LGTM

File: config/crd/bases/service.binding_servicebindings.yaml (hunk 1 above)
Comment (baijum) [reply config/crd/bases/service.binding_servicebindings.yaml:7]: Created PR to add the check: #20

**Pull request merged**